src/main/java/net/wasdev/wlp/common/springboot/util/SpringBootThinUtil.java

```diff
@@ -20,183 +25,433 @@
+        while ((read = eis.read(buffer)) != -1) {
+            digest.update(buffer, 0, read);
+        }
+        byte[] digested = digest.digest();
+        return convertToHexString(digested);
+    }
+
+    private static String convertToHexString(byte[] digested) {
+        StringBuilder stringBuffer = new StringBuilder();
+        for (int i = 0; i < digested.length; i++) {
+            stringBuffer.append(Integer.toString((digested[i] & 0xff) + 0x100, 16).substring(1));
+        }
+        return stringBuffer.toString();
+    }
+
+    private void storeLibraryInDir(JarEntry entry, String hashPrefix, String hashSuffix) throws IOException, NoSuchAlgorithmException {
+        String hashPath = hashPrefix + '/' + hashSuffix;
+        String libName = entry.getName();
+        int lastSlash = libName.lastIndexOf('/');
+        if (lastSlash >= 0) {
+            libName = libName.substring(lastSlash + 1);
+        }
+
+        if (libIndexCacheParent != null) {
+            // if there is a parent cache look to see if the lib name exists there
+            File libDirParent = new File(libIndexCacheParent, hashPath);
+            File libFileParent = new File(libDirParent, libName);
+            if (libFileParent.exists()) {
+                // no need to store since the lib exists in the parent cache
+                return;
+            }
+        }
+
+        if (!libIndexCache.exists()) {
+            libIndexCache.mkdirs();
+        }
+
+        File libDir = new File(libIndexCache, hashPath);
+        File libFile = new File(libDir, libName);
+        if (!libFile.exists()) {
+            if (!libDir.exists()) {
+                libDir.mkdirs();
+            }
+            InputStream is = sourceFatJar.getInputStream(entry);
+            try (OutputStream libJar = new FileOutputStream(libFile)) {
+                copyStream(is, libJar);
+            } finally {
+                is.close();
+            }
+        }
+    }
 
-	private void addLibIndexFileToThinJar(JarOutputStream thinJar) throws IOException {
-		thinJar.putNextEntry(new ZipEntry(SPRING_LIB_INDEX_FILE));
-		try {
-			for (String libEntry : libEntries) {
-				thinJar.write(libEntry.getBytes(StandardCharsets.UTF_8));
-				thinJar.write('\n');
-			}
-		} finally {
-			thinJar.closeEntry();
-		}
+    private void writeEntry(InputStream is, ZipOutputStream zos, String entryName) throws IOException {
+        try {
+            zos.putNextEntry(new ZipEntry(entryName));
+            copyStream(is, zos);
+        } finally {
+            zos.closeEntry();
+        }
+    }
+
+    private void copyStream(InputStream is, OutputStream os) throws IOException {
+        byte[] buffer = new byte[4096];
+        int read = -1;
+        while ((read = is.read(buffer)) != -1) {
+            os.write(buffer, 0, read);
+        }
+    }
+
+    private void addLibIndexFileToThinJar(JarOutputStream thinJar) throws IOException {
+        thinJar.putNextEntry(new ZipEntry(SPRING_LIB_INDEX_FILE));
```

Comment:
Should this be inside the try block?

Reply:
addLibIndexFileToThinJar is called inside the try block.